vars/randomString.groovy

```diff
@@ -0,0 +1,36 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+Generate a random string
+
+  // Create a random string of 15 chars (alphanumeric)
+  def randomUuid = uuid(size: 15)
+*/
+
+import java.security.SecureRandom
+import java.util.Base64
+
+def call(Map args = [:]) {
+  def size = args.get('size', 128)
+  SecureRandom rand = new SecureRandom()
+  byte[] randomBytes = new byte[256]
+  rand.nextBytes(randomBytes)
+  // alphanumeric and dash are allowed but not ending with dash
+  def value = new String(Base64.encoder.encode(randomBytes)).replaceAll("[\\W]|-", "-").take(size).replaceAll('-$', 'a')
```

Comment:
we have an step to encode/ecode base64

https://github.com/elastic/apm-pipeline-library/blob/master/vars/base64decode.groovy
https://github.com/elastic/apm-pipeline-library/blob/master/vars/base64encode.groovy

Reply:
They consume a string, in this case it requires to consume the bytes to generate a random string

Reply:
also, a base64 string can only have `0-9`, `A-Z`, and `=`, I think you are expecting a UUID

Reply:
not anymore an UUID but a random string, but somehow the base64 was generating some other chars:
for instance:

Given:
```
[79, 56, -33, 83, 83, -124, 33, -81, 69, 90, 111, -70, -2, -85, 100, -26, 98, -6, 85, -48, 22, 54, -109, 37, -44, 116, -22, 14, -27, 45, 99, 24, -1, -95, 50, -21, 35, -77, -57, -40, -70, -37, 71, 95, 37, 89, -34, -123, -88, -91, 70, 48, 55, -26, 107, -81, -124, 71, 7, 101, -120, -127, -8, -78, -121, 108, 53, -27, -110, 104, -63, 7, -69, -28, 64, 74, -38, -50, 108, 98, 26, 71, -23, 30, -98, 107, -121, -96, -6, -122, 103, 119, 42, -113, -29, 105, -105, -87, 11, 33, -93, -87, -2, 119, -42, -123, 9, 77, -9, -109, -24, -88, 4, 47, 127, 110, 95, 100, 12, 83, -79, -112, -91, -67, -104, 1, -74, -6]
```

The output was:
```
Ia9FWm+6/qtk5mL6VdAWNpMl1HTqDuUtYxj/oTLrI7PH2LrbR18lWd6FqKVGMDfma6+ERwdliIH4sodsNeWSaMEHu+RAStrObGIaR+kenmuHoPqGZ3cqj+Npl6kLIaOp/nfWhQlN95PoqAQvf25fZAxTsZClvZgBtvo=
```

Reply:
there are two characters more allow on base 64 `/` and `+` I didn't remember it
https://en.wikipedia.org/wiki/Base64